stdlib/Random/src/generation.jl

```diff
@@ -210,7 +210,7 @@ SamplerRangeFast(r::AbstractUnitRange{T}) where T<:BitInteger =
     SamplerRangeFast(r, uint_sup(T))
 
 function SamplerRangeFast(r::AbstractUnitRange{T}, ::Type{U}) where {T,U}
-    isempty(r) && throw(ArgumentError("range must be non-empty"))
+    isempty(r) && throw(ArgumentError("collection must be non-empty"))
```

Comment:
This fixes the problem when the array has indexes of type `Int128`, but not the main case of `Array`. Other instances of `"range must be non-empty"` should also be updated.